src/components/AddComment.jsx

```diff
@@ -0,0 +1,61 @@
+import { useState } from "react";
+import "./AddComment.css";
+import { addArticleComment } from "../api";
+
+const AddComment = ({ article_id, currentUser, comments, setComments }) => {
+  const [commentContent, setCommentContent] = useState("");
+  const [notification, setNotification] = useState(null);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    const comment_id = Math.max(...comments.map((c) => c.comment_id)) + 1;
+    const newComment = {
+      comment_id,
+      article_id,
+      author: currentUser,
+      body: commentContent,
+      created_at: new Date(),
+      votes: 0,
+    };
+
+    setComments((current) => [newComment, ...current]);
+
+    addArticleComment(article_id, {
+      username: currentUser,
+      body: commentContent,
+    })
+      .then((_) => {
+        setNotification("Thanks for your comment!");
+        setTimeout(() => setNotification(null), 2000);
+      })
+      .catch((_) => {
+        setComments([...comments]);
+        setNotification("Something went wrong...");
+        setTimeout(() => setNotification(null), 5000);
+      });
+
+    setCommentContent("");
+  };
+
+  return notification ? (
```

Comment:
I like how you used a timeout for this! however, think of web design conventions - do you know many websites where they replace the comment box with a message after something is posted? Instead, I would consider making a notification element that renders alongside the comment box, and then use your timeout that way. You can still disable the comment form another way to prevent users spamming if you want to.